lib/model/model.go

```diff
@@ -2444,14 +2454,11 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 }
 
 func (m *model) GetFolderVersions(folder string) (map[string][]versioner.FileVersion, error) {
-	fcfg, ok := m.cfg.Folder(folder)
-	if !ok {
-		return nil, errFolderMissing
-	}
-
-	ver := fcfg.Versioner()
+	m.fmut.RLock()
+	ver := m.folderVersioners[folder]
```

Comment:
I'd keep returning `errFolderMissing` if `!ok` (using `ver, ok := ...`) - it's a tiny bit more accurate.